### PR TITLE
feat(grouping): New iteration of the newstyle trimming with message stripping

### PR DESCRIPTION
**Pull request by mitsuhiko**
This is a work in progress new grouping algorithm based on the newstyle one
which re-introduces exception value when stacktraces are not available but it
strips out a bunch of values which are not intended to appear in messages.

This is based on ted's suggestion for removing values from messages which are
likely to not be helpful for grouping.

The plan is for this to be iterated on until we feel good about using it for javascript
and to then freeze it. Since this also changes the message grouping I opted for
making it a new version.

### CODE REVIEW

File: tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/newstyle:2019_04_17/javascript_exception_fallback_to_message_whistles.pysnap [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+---
+app:
+  hash: None
+  component:
+    app (exception of system takes precedence)
+      exception (ignored because hash matches system variant)
+        type*
+          u'Error'
+        value* (stripped common values)
+          u'<date>: <email> logged in (error <int>) time spent <float> --- correlation id <uuid>, checksum <sha1> (md5 <md5>); payload timestamp <date>'
```
Comment (mitsuhiko) [inline tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/newstyle:2019_04_17/javascript_exception_fallback_to_message_whistles.pysnap]: The input string for reference is:

```
Wed Apr 17 22:31:45 2019: foobar.bazblafasel@example.invalid logged in (error 42) time spent 1234.33 --- correlation id 88ee7c66-29a9-4a4a-bc26-c72d76c5de14, checksum da39a3ee5e6b4b0d3255bfef95601890afd80709 (md5 d41d8cd98f00b204e9800998ecf8427e); payload timestamp 2019-04-17T20:29:05Z
```

Comment (mitsuhiko) [reply tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/newstyle:2019_04_17/javascript_exception_fallback_to_message_whistles.pysnap]: @tkaemming thoughts? this is based on a conversation we had about grouping and similarity a few weeks back when applied to fallback messages.

File: src/sentry/grouping/strategies/message.py [hunk 2]
```diff
@@ -1,9 +1,82 @@
...
+            [0-9a-fA-F]{4}-
+            [0-9a-fA-F]{12}
+        \b
+    ) |
+    (?P<sha1>
+        \b[0-9a-fA-F]{40}\b
+    ) |
+    (?P<md5>
+        \b[0-9a-fA-F]{32}\b
+    ) |
```
Comment (markstory) [inline src/sentry/grouping/strategies/message.py]: Should IPv4 patterns be included in this noisy data list?

Comment (mitsuhiko) [reply src/sentry/grouping/strategies/message.py]: Indeed, that sounds like a good idea.

File: tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/newstyle:2019_04_17/javascript_exception_fallback_to_message.pysnap [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+---
+app:
+  hash: None
+  component:
+    app (exception of system takes precedence)
+      exception (ignored because hash matches system variant)
+        type*
+          u'Error'
+        value* (stripped common values)
+          u'Loading chunk <int> failed.\n(timeout: <url>'
```
Comment (jan-auer) [inline tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/newstyle:2019_04_17/javascript_exception_fallback_to_message.pysnap]: This looks a bit strange as it strips the parenthesis, but it's unlikely for this to ever produce false positives.

**Review by jan-auer — APPROVED** [review]

File: tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/newstyle:2019_04_17/javascript_exception_fallback_to_message_whistles.pysnap (hunk 1 above)
Comment (tkaemming) [reply tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/newstyle:2019_04_17/javascript_exception_fallback_to_message_whistles.pysnap]: +1, this is great

**Pull request merged**